mat-process-api/V1/Gateways/ProcessDataGateway.cs

```diff
@@ -27,7 +28,24 @@ public MatProcessData GetProcessData(string processRef)
             var filter = Builders<BsonDocument>.Filter.Eq("_id", processRef);
             //we will never expect more than one JSON documents matching an ID so we always choose the first/default result
             var result = matDbContext.getCollection().FindAsync(filter).Result.FirstOrDefault();
+            
+            return ProcessDataFactory.CreateProcessDataObject(result);
+        }
 
+        public MatProcessData UpdateProcessData(UpdateDefinition<BsonDocument> updateDefinition, string id)
+        {
+            //return updated document
+            var options = new FindOneAndUpdateOptions<BsonDocument>();
+            options.ReturnDocument = ReturnDocument.After;
+            //query by
+            var filter = Builders<BsonDocument>.Filter.Eq("_id", id);
+            //find and update
+            var result = matDbContext.getCollection().FindOneAndUpdate(filter, updateDefinition, options);
+            //if empty result, the document to be updated was not found
+            if(result == null)
```

Comment:
Not sure how this works - doesn't the attempt at update against non-existent resource make mongo driver throw an error? Or does it just return 'null' if the attempt fails?

Reply:
If document doesn't exist, it's null it returns